ipn/ipnlocal/local.go

```diff
@@ -1703,6 +1703,11 @@ func applySysPolicy(prefs *ipn.Prefs, lastSuggestedExitNode tailcfg.StableNodeID
 		anyChange = true
 	}
 
+	if hostname, err := syspolicy.GetString(syspolicy.Hostname, prefs.Hostname); len(hostname) > 0 && err == nil && prefs.Hostname != hostname {
```

Comment:
I think comparing to `""` is generally preferred in Go, and there's probably no reason to pass `prefs.Hostname` as the default value. Maybe this?
```suggestion
	if hostname, _ := syspolicy.GetString(syspolicy.Hostname, ""); hostname != "" && hostname != prefs.Hostname {
```

Reply:
Or, actually, maybe unset the preference if the policy setting no longer applies?

Reply:
It is my understanding that we let the user set the hostname themselves in the preferences, e.g. by providing the `--hostname` argument to `up` or `set`. So I don't think we should ever unset the preference; we should only override the preference value if the MDM is giving us a value to override with, and leave everything else unchanged.

Reply:
Consider this scenario: a company configures a hostname policy setting for their devices but later needs to revert the change for any reason. Currently, there's no way to do this without instructing users to manually reset it through the CLI.

An acceptable temporary solution would be to allow specifying an empty string as the policy value, which would reset `Prefs.Hostname` to `""` (and likely set `tailcfg.Hostinfo.Hostname` back to `os.Hostname()`). This still isn't ideal, as the common expectation for policies is to revert behavior when a policy setting no longer applies. Unfortunately, this isn't always the case for Tailscale policies that modify prefs (I have a WIP to improve this). However, it at least offers a usable workaround for admins.

Reply:
Ha, I missed that `setPrefs()` gets called later in the flow and that will persist the values to disk. I assumed we were just changing the value _in-memory_. I'll make it accept an empty string as you suggested, it seems like a reasonable behavior to me.